### PR TITLE
JP-2343: Add CRDS context log message to JwstStep

### LINKED ISSUE #6486: Step/Pipelines should log the CRDS context
_Issue [JP-2343](https://jira.stsci.edu/browse/JP-2343) was created on JIRA by [Jonathan Eisenhamer](https://jira.stsci.edu/secure/ViewProfile.jspa?name=eisenham):_

Steps/Pipelines should log the CRDS context being used.

Note that there is a verbosity factor involved: Ideally all Steps and Pipelines that use CRDS should log this info. However, when running a Pipeline, that will be quite a bit of the same info being logged, as the Pipeline and all executed Steps will be doing so. Consider methods of mitigation.

### PR DESCRIPTION
**Pull request by tapastro**
Closes #6486
Resolves [JP-2343](https://jira.stsci.edu/browse/JP-2343)

**Description**

This PR adds a log message to report the CRDS context by having JwstPipeline report in finalize_result. It adds the same log in JwstStep finalize_result, but is only provided if the step has no parent, i.e. standalone step running vs. a step within a pipeline.

Checklist
- [ ] Tests
- [ ] Documentation
- [x] Change log
- [x] Milestone
- [x] Label(s)

### CODE REVIEW

File: jwst/stpipe/core.py [hunk 1]
```diff
@@ -11,7 +11,10 @@
 from stpipe import Step
 from stpipe import Pipeline
 from ..lib.suffix import remove_suffix
+import logging
 
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
```
Comment (stscieisenhamer) [inline jwst/stpipe/core.py:17]: Why is the level being set?

**Review by stscieisenhamer — COMMENTED** [review]
Besides changelog, LGTM.

Beautiful!

File: jwst/stpipe/core.py (hunk 1 above)
Comment (tapastro) [reply jwst/stpipe/core.py:17]: Good question - it was a brainless copy-paste from a different step, but I don't think it's needed anywhere. It does exist in nearly every python file that gets imported into a _step.py file, though. Could be something to cleanse the pipeline of, in another PR.

Comment (stscieisenhamer) [reply jwst/stpipe/core.py:17]: Ah right. There is actually an issue about this: [JP-1090](https://jira.stsci.edu/browse/JP-1090)

**Review by stscieisenhamer — APPROVED** [review]

**Review by hbushouse — APPROVED** [review]
LGTM

**Comment by tapastro** [comment]
@cshanahan1 Nadia asked me to ping you on this PR for Roman purposes - I don't know if it'll be helpful or not 😬

**Comment by hbushouse** [comment]
flake8 is not happy with you

**Pull request merged**